### DISCUSSION TITLE
How can I change default 8443 port to 443

### DISCUSSION THREAD

**Discussion by tharukan** [keycloak/keycloak#13375] [Q&A]
Hi. Can I change the default 8443 port to 443. For an example, when keycloak loading URL is like this.
_**https://keycloak.local:8443**_
I just wanted to load the keycloak without any port showing like _**https://keycloak.local**_. I'm not using any reverse proxy. 
Request your support on this. Thanks.

**Comment by georgylazarev** [comment]
Just go to the `keycloak.conf` file and add 
`https-port=443`

**Reply by setonix-pratik** [reply]
I tried this but got the error ` ERROR: io.quarkus.runtime.QuarkusBindException: Port(s) already bound: 443: Permission denied`

while there's nothing setup within the port

**Comment by darius-m** [comment]
Ports under 1024 (1-1023) are protected, since most common services (FTP, SSH, telnet, HTTP, HTTPS, etc.) use these ports. As such, you cannot usually bind to these ports as a normal user. You can either use a `sysctl` to change the range of protected ports (not usually recommended) or use an application that can start as `root` and then drop privileges (i.e., switch user, SELinux context, capabilities). Since Keycloak itself does not have this functionality (as far as I know), you will likely need to use a reverse proxy.

**Comment by mlromramse** [comment]
One solution I found attractive is to set a port-forward in the firewall:
    `sudo firewall-cmd --zone=public --add-masquerade --permanent`
    `sudo firewall-cmd --zone=public --add-forward-port=port=443:proto=tcp:toport=8443 --permanent`

**Comment by mlromramse** [comment]
Another solution is to run keycloak in a container (docker|podman) and expose port 8443 from the container as 443:
`sudo podman run -d -p 443:8443 quay.io/keycloak/keycloak:20.0.5 start-dev`

**Comment by edenallen7** [comment]
You can’t directly switch Keycloak’s default 8443 to 443 without root privileges; bind Keycloak to [443](https://cheapsslweb.com/blog/port-443-https/) or use iptables/firewall rules for port forwarding if no reverse proxy is used.
